department-of-veterans-affairs/next-build
# VACMS-14824: Pagination for story listing pages

## Description

Relates to https://github.com/department-of-veterans-affairs/va.gov-cms/issues/14824

Adds pagination to story-listing pages by statically generating additional pages according to the total number of stories and page size. For example:
- Listing page `some-health-care-system/stories` contains 25 stories.
- Page size is 10 stories.
- `some-health-care-system/stories` is kept as first page, and `some-health-care-system/stories/page-2` and `some-health-care-system/stories/page-3` are added.

## QA steps
Note: this will likely fail locally. This does not mean something with this PR is the issue. Rather, this PR introduces the need to hit Drupal with additional requests, and Drupal is falling down as a result sometimes.
1. On Tugboat:
    - [x] Validate that build succeeds
    - [x] Validate the existence of some paginated listing pages
       - [x] `eastern-oklahoma-health-care/stories/` should exist
       - [x] `eastern-oklahoma-health-care/stories/page-2` should exist
       - [x] `eastern-oklahoma-health-care/stories/page-7` should exist
       - [x] `eastern-oklahoma-health-care/stories/page-8` should NOT exist

2. If run locally, you can slice the data results to deal with a smaller subset (see below):
    - [x] Validate that build succeeds
    - [x] Validate that the `out` directory contains some paginated listing pages
       - [x] `eastern-oklahoma-health-care/` should exist
       - [x] `eastern-oklahoma-health-care/stories/page-2` should exist
       - [x] `eastern-oklahoma-health-care/stories/page-7` should exist
       - [x] `eastern-oklahoma-health-care/stories/page-8` should NOT exist
```
/* src/lib/utils/staticPaths.ts */
export async function getStaticPathsByResourceType(
  resourceType,
  context: GetStaticPathsContext
): ReturnType<typeof drupalClient.getStaticPathsFromContext> {
  return (await drupalClient.getStaticPathsFromContext([resourceType], context)).slice(0, 5)
}
```
